src/robot/result/keywordremover.py

```diff
@@ -172,6 +173,13 @@ def _remove_keywords(self, body):
                     body.remove(kw)
 
 
+class NotRunKeywordRemover(KeywordRemover):
+    def visit_test(self, test):
+        before = len(test.body)
+        test.body = [kw for kw in test.body if kw.status != "NOT RUN"]
+        self.removal_message.set_to_if_removed(test, before)
```

Comment:
This only removes keywords from test body, not from child keywords or control structures. The easiest way to extend the functionality would be implementing `start_test` instead of `visit_test` (that way also body is visited automatically) and implementing also `start_body_item` that's, by default, called with all keywords and control structures.